packages/hathor-rpc-handler/src/rpcMethods/createToken.ts

```diff
@@ -26,7 +26,16 @@ import { z } from 'zod';
 const createTokenSchema = z.object({
   name: z.string().min(1),
   symbol: z.string().min(1),
-  amount: z.number().positive(),
+  amount: z.string().refine(val => {
+    try {
+      const bigIntVal = BigInt(val);
+      return bigIntVal > BigInt(0);
+    } catch (e) {
+      return false;
+    }
+  }, {
+    message: "Amount must be a valid positive number string"
+  }).transform(val => BigInt(val)),
```

Comment:
suggestion: Use the coerce which will output a bigint.

```suggestion
  amount: z.union([z.bigint(), z.number(), z.string().regex(/^\d+$/)])
    .pipe(z.coerce.bigint().positive()),
```

Or if you want to change the error message with refine.
```ts
z.union([z.bigint(), z.number(), z.string().regex(/^\d+$/)])
  .pipe(z.coerce.bigint())
  .refine(val => val > 0n, "Amount must be a valid positive number.")
```

Obs: If you want to enable points and commas in the string version you will need to change the regex to something like `/^(\d[\.,]?)*\d$/`

Reply:
That's a great suggestion, thanks!

I refactored the code, but removed `z.bigint()` and `z.number()` as we will only accept number strings for now